foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/helper/ConcurrencyUtil.java

```diff
@@ -0,0 +1,1490 @@
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Oracle - initial API and implementation from Oracle TopLink
+package org.eclipse.persistence.internal.helper;
+
+import org.eclipse.persistence.config.SystemProperties;
+import org.eclipse.persistence.internal.helper.type.CacheKeyToThreadRelationships;
+import org.eclipse.persistence.internal.helper.type.ConcurrencyManagerState;
+import org.eclipse.persistence.internal.helper.type.DeadLockComponent;
+import org.eclipse.persistence.internal.helper.type.ReadLockAcquisitionMetadata;
+import org.eclipse.persistence.internal.identitymaps.CacheKey;
+import org.eclipse.persistence.internal.localization.TraceLocalization;
+import org.eclipse.persistence.internal.security.PrivilegedAccessHelper;
+import org.eclipse.persistence.internal.security.PrivilegedGetSystemProperty;
+import org.eclipse.persistence.logging.AbstractSessionLog;
+import org.eclipse.persistence.logging.SessionLog;
+
+import java.io.StringWriter;
+import java.lang.management.ManagementFactory;
+import java.lang.management.ThreadInfo;
+import java.lang.management.ThreadMXBean;
+import java.security.AccessController;
+import java.util.*;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class ConcurrencyUtil {
+
+    public static final ConcurrencyUtil SINGLETON = new ConcurrencyUtil();
```

Comment:
Is this supposed to be real singleton or is it the variable name causing confusion on my end?

Reply:
Yes it's real singleton. Private constructor was added.